wire-ios/Wire-iOS Tests/ConversationCreation/GroupConversationCreationCoordinatorTests.swift

```diff
@@ -0,0 +1,585 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+@testable import Wire
+
+class MockConversationCreator: GroupConversationCreator {
+    var conversation: ZMConversation?
+
+    init() {
+    }
+
+    func insertGroupConversation(moc: NSManagedObjectContext, participants: [ZMUser], name: String?, team: WireDataModel.Team?, allowGuests: Bool, allowServices: Bool, readReceipts: Bool, messageProtocol: WireDataModel.MessageProtocol) -> ZMConversation? {
+        conversation = ZMConversation.insertGroupConversation(
+            moc: moc,
+            participants: participants,
+            name: name,
+            team: team,
+            allowGuests: allowGuests,
+            allowServices: allowServices,
+            readReceipts: readReceipts,
+            messageProtocol: messageProtocol
+        )
+        return conversation
+    }
+}
+
+final class GroupConversationCreationCoordinatorTests: XCTestCase {
+    var sut: GroupConversationCreationCoordinator!
+    var coreDataStack: CoreDataStack!
+    var uiMOC: NSManagedObjectContext!
+    var documentsDirectory: URL?
+
+    override func setUp() {
+        super.setUp()
+        do {
+            documentsDirectory = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+        } catch {
+            XCTAssertNil(error, "Unexpected error \(error)")
+        }
+        setupCoreDataStack()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        sut?.finalize()
+        // Needs to be called before setting self.documentsDirectory to nil.
+        removeContentsOfDocumentsDirectory()
+        uiMOC = nil
+        coreDataStack = nil
+        documentsDirectory = nil
+    }
+
+    func setupCoreDataStack() {
+        let account = Account(userName: "", userIdentifier: UUID())
+        let coreDataStack = CoreDataStack(account: account,
+                                          applicationContainer: documentsDirectory!,
+                                          inMemoryStore: true)
+
+        coreDataStack.loadStores(completionHandler: { error in
+            XCTAssertNil(error)
+        })
+        self.coreDataStack = coreDataStack
+        self.uiMOC = coreDataStack.viewContext
```

Comment:
You may find there exists already a test class that has this setup logic.

Reply:
It is a class for snapshot tests, should I use this one, or create a new base class?

Reply:
It shouldn't be a snapshot test, so I'd suggest a new base class.

Reply:
I've used `CoreDataFixtureTestHelper` instead